controllers/quote.js

```diff
@@ -157,37 +172,103 @@ exports.get = function *() {
         'source transfers')
     }
   } else {
-    sourceExpiryDuration = destinationExpiryDuration +
-      config.expiry.minMessageWindow
+    sourceExpiryDuration = destinationExpiryDuration + config.expiry.minMessageWindow
+  }
+
+  let source_ledger = params.source_ledger
+  if (!source_ledger) {
+    if (params.source_account) source_ledger = yield getAccountLedger(params.source_account)
+    else throw new Error('Missing required parameter: source_ledger or source_account')
   }
 
-  let paymentTemplate = {
+  let destination_ledger = params.destination_ledger
+  if (!destination_ledger) {
+    if (params.destination_account) destination_ledger = yield getAccountLedger(params.destination_account)
+    else throw new Error('Missing required parameter: destination_ledger or destination_account')
+  }
+
+  return {
+    destinationExpiryDuration: destinationExpiryDuration,
+    sourceExpiryDuration: sourceExpiryDuration,
+    source_amount: params.source_amount,
+    destination_amount: params.destination_amount,
+    source_ledger: source_ledger,
+    destination_ledger: destination_ledger,
+    source_account: params.source_account || null,
+    destination_account: params.destination_account || null
+  }
+}
+
+function makeQuoteArgs (query) {
+  return {
+    source_ledger: query.source_ledger,
+    destination_ledger: query.destination_ledger,
+    source_amount: query.source_amount,
+    destination_amount: query.destination_amount
+  }
+}
+
+function makePaymentTemplate (query, quote) {
+  const source_amount = quote.source_amount.toFixed(2, 2)
```

Comment:
This is probably a sane default for now, but will probably have to change in the future. The decimal places should probably be based on the precision that the various backends support.